apps/examples/smartfs/buffer_optimization/buffer_optimization_main.c

```diff
@@ -0,0 +1,449 @@
+ * Name: create_test_file
+ *
+ * Description: Creates the test file with test data that we will use
+ *              to conduct the test.
+ *
+ ****************************************************************************/
+
+static int create_test_file(int f_size)
+{
+	int fd;
+	int ret;
+	memset(g_buf, 1024, '1');
+
+	fd = open(SAMPLE_FILE, O_WROK | O_CREAT);
+	if (fd < 0) {
+		printf("Unable to open Sample File for testing, Error = %d\n", errno);
+		return ERROR;
+	}
+
+	while (f_size >= 1024) {
+		ret = write(fd, g_buf, 1024);
+		if (ret != 1024) {
+			printf("Unable to write to Sample File, Error = %d\n", errno);
+			return ERROR;
+		}
+		f_size -= 1024;
+	}
+	if (f_size) {
+		ret = write(fd, g_buf, f_size);
+		if (ret != f_size) {
+			printf("Unable to write to Sample File, Error = %d\n", errno);
+			return ERROR;
+		}
+	}
+	close(fd);
+
+	return OK;
+}
+
+/****************************************************************************
+ * Name: append_test
+ *
+ * Description: Perform appending of data to the file, buffer by buffer
+ *
+ ****************************************************************************/
+
+static int append_test(int size)
+{
+	int ret;
+	int fd;
+	int itr = 0;
+	int total = TOTAL_APPEND_SIZE;
+
+	while (itr < APPEND_TEST_ITR) {
+		fd = open(SAMPLE_FILE, O_WROK | O_APPEND);
+		if (fd < 0) {
+			printf("Unable to open sample file for appending, Errno = %d\n", errno);
+			return ERROR;
+		}
+
+		while (total >= size) {
+			ret = write(fd, g_buf, size);
+			if (ret != size) {
+				printf("Unable to write to file, ret = %dbytes\n", ret);
+				return -EIO;
+			}
+			total -= size;
+		}
+		if (total) {
+			ret = write(fd, g_buf, total);
+			if (ret != total) {
+				printf("Unable to write to file, ret = %dbytes\n", ret);
+				return -EIO;
+			}
+		}
+		close(fd);
+		itr += 1;
+	}
+
+	return OK;
+}
+
+/****************************************************************************
+ * Name: append_test_util
+ *
+ * Description: Set up the timer and run the append test for different
+ *              buffer sizes.
+ *
+ ****************************************************************************/
+
+static int append_test_util(int frt_fd, int size)
+{
+	int ret;
+	int buf_size;
+	struct timer_status_s time_1, time_2;
+	uint32_t time_append;
+	struct stat st;
+
+	if (ioctl(frt_fd, TCIOC_GETSTATUS, (unsigned long)(uintptr_t)&time_1) < 0) {
+		fprintf(stderr, "ERROR: Failed to get Free Run Timer status: %d\n", errno);
+		return -1;
+	}
+
+	ret = append_test(512);
+	if (ret != OK) {
+		printf("Appending with buffer size 512 failed\n");
+		return ret;
+	}
+
+	if (ioctl(frt_fd, TCIOC_GETSTATUS, (unsigned long)(uintptr_t)&time_2) < 0) {
+		fprintf(stderr, "ERROR: Failed to get Free Run Timer status: %d\n", errno);
+		return -1;
+	}
+
+	time_append = time_2.timeleft - time_1.timeleft;
+	print_test_result("appending", size, APPEND_TEST_ITR, 512, time_append);
```

Comment:
To calculate the throughput of append operation, you need to use TOTAL_APPEND_SIZE not size.